### PR TITLE
Add Amazon Personalize Provider

### PR DESCRIPTION
**Pull request by zhashuyi**
Add Amazon Personalize Provider

Related to https://github.com/aws-amplify/amplify-js/pull/2915

### CODE REVIEW

File: js/analytics.md [hunk 1]
```diff
@@ -336,6 +336,188 @@ Analytics.record({
...
+Analytics.record({
+    eventType: "MediaAutoTrack",
+    userId: "<USER_ID>", (optional)
+    properties: {
+        "domElementId": "MEDIA DOM ELEMENT ID",
+        "itemId": "<ITEM_ID>"
+    }
+}, "AmazonPersonalize");
+```
+## Using Amazon Personalize
```
Comment (elorzafe) [inline js/analytics.md]: It seems duplicated from the top one

Comment (zhashuyi) [reply js/analytics.md]: Removed the duplicated one.

File: js/analytics.md [hunk 2]
```diff
@@ -336,6 +336,97 @@ Analytics.record({
...
+```javascript
+
+// Configure the plugin after adding it to the Analytics module
+Analytics.configure({
+    AmazonPersonalize: {
+    
+        // REQUIRED - The trackingId to track the events 
+        trackingId: '<TRACKING_ID>',
+        
+        // OPTIONAL -  Amazon Kinesis service region
```
Comment (sepehr) [inline js/analytics.md]:
```js
        // OPTIONAL -  Amazon *Personalize* service region
```

Comment (zhashuyi) [reply js/analytics.md]: Thanks Resolved

**Pull request merged**